wordpress-mobile/WordPress-Android
# display error screen for webviews while offline or with poor connection

Fixes #9563 substask display error screen when user tries to "view" a post while offline

This PR makes it so that if a user tries to "view" a published post while being offline or with poor internet connection, he or she is taken to an error screen with a retry button. Once the user gets data and he or she clicks on the "retry" button he or she should be taken to the actual webview, if he or she has good connection.  This will also affect other places in the app that behave the same way.

![offlineview](https://user-images.githubusercontent.com/4370019/57588691-ed5ae280-74e5-11e9-8341-525a5f16079b.gif)

To test:
1.Turn on airplane mode
2.Go to the "published" posts tab. 
3.press "view" on the post while offline or with poor internet connection.
4.you should be taken to an error screen with a "retry" button
5.Turn off airplane mode
6.When you get data proceed to click on the "retry" button.

Unit test:
run WPWebViewViewModelTest

P.S: 
Currently the error screen only contains a retry button. 
I will be creating another PR which will contain mostly UI changes for the error screen (error message, error image, etc), and the progress screen, and I also set up a little bit of time delays from switching to progress screen to error screen, or progress screen to success screen and so on.
